test/test_helper.rb

```diff
@@ -185,5 +185,17 @@ def stub_hostip_requests
       stub_request(:get, "http://api.hostip.info/country.php?ip=0.0.0.0")
       stub_request(:get, "http://api.hostip.info/country.php?ip=127.0.0.1")
     end
+
+    def email_text_parts(message)
+      text_parts = []
+      message.parts.each do |part|
```

Comment:
You could avoid initialising the array and then returning it by using `each_with_object` here. So something like:

```
message.parts.each_with_object([]) do |part,text_parts|
```

and then get rid of the first and last lines of the method.